sdk/src/main/java/com/mapbox/maps/ViewAnnotationManagerImpl.kt

```diff
@@ -143,6 +150,157 @@ internal class ViewAnnotationManagerImpl(
+    annotations: List<View>,
+    edgeInsets: EdgeInsets?,
+    bearing: Double?,
+    pitch: Double?,
+    animationDurationMs: Long
+  ) {
+    if (annotations.isNotEmpty()) {
+      val viewAnnotationOptions = annotations.mapNotNull {
+        getViewAnnotationOptionsByView(it)
+      }
+      val coordinateBounds = coordinateBoundsFromAnnotations(viewAnnotationOptions)
+      val paddings = calculateEdgeInset(viewAnnotationOptions, edgeInsets)
+      val geometry = GeometryCollection.fromGeometries(
+        listOf(
+          coordinateBounds.northwest(),
+          coordinateBounds.northeast,
+          coordinateBounds.southwest,
+          coordinateBounds.southeast()
+        )
+      )
+      val overviewViewportState = mapView.viewport.makeOverviewViewportState(
+        OverviewViewportStateOptions.Builder()
+          .geometry(geometry)
+          .padding(paddings)
+          .animationDurationMs(animationDurationMs)
+          .build()
+      )
+      mapView.viewport.transitionTo(
+        targetState = overviewViewportState,
+        completionListener = { success ->
+          if (success) {
+            logI("ViewAnnotation", "transition to annotations success")
+          } else {
+            logI("ViewAnnotation", "transition to annotations failed")
+          }
+        })
+    }
+  }
+
+  override fun cameraForAnnotations(
+    annotations: List<View>,
+    edgeInsets: EdgeInsets?,
+    bearing: Double?,
+    pitch: Double?
+  ): CameraOptions? {
+    if (annotations.isNotEmpty()) {
+      val viewAnnotationOptions = annotations.mapNotNull {
+        getViewAnnotationOptionsByView(it)
+      }
+      val coordinateBounds = coordinateBoundsFromAnnotations(viewAnnotationOptions)
+      val paddings = calculateEdgeInset(viewAnnotationOptions, edgeInsets)
+      return mapboxMap.cameraForCoordinateBounds(
+        coordinateBounds,
+        paddings,
+        bearing,
+        pitch
+      )
+    }
+    return null
+  }
+
+  /**
+   * Convenience function to get bounds occupied by list of [ViewAnnotationOptions]'s geometry.
+   */
+  private fun coordinateBoundsFromAnnotations(annotationOptions: List<ViewAnnotationOptions>): CoordinateBounds {
+    var north = -MAX_LATITUDE
+    var south = MAX_LATITUDE
+    var west = MAX_LONGITUDE
+    var east = -MAX_LONGITUDE
+    annotationOptions.mapNotNull {
+      it.geometry?.let { geometry ->
+        val point = Point.fromJson(geometry.toJson())
+        north = maxOf(north, point.latitude())
+        south = minOf(south, point.latitude())
+        west = minOf(west, point.longitude())
+        east = maxOf(east, point.longitude())
+      }
+    }
+    return CoordinateBounds(Point.fromLngLat(west, south), Point.fromLngLat(east, north), false)
+  }
+
+  /**
+   * Calculate paddings to show viewAnnotations.
+   * Get the topMost, leftMost, rightMost, bottomMost annotation options and apply paddings accordingly.
+   */
+  private fun calculateEdgeInset(
```

Comment:
Would be nice to drop some tests for the calculation methods

Reply:
Yes, I will add tests.